File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/powergrid/android/PowerGridMainActivity.java
package com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.android;

import com.zwstudio.logicpuzzlesandroid.R;
import com.zwstudio.logicpuzzlesandroid.common.android.GameMainActivity;
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.data.PowerGridDocument;
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.domain.PowerGridGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.domain.PowerGridGameMove;
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.domain.PowerGridGameState;

import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;

@EActivity(R.layout.activity_game_main)
public class PowerGridMainActivity extends GameMainActivity<PowerGridGame, PowerGridDocument, PowerGridGameMove, PowerGridGameState> {
    @Bean
    protected PowerGridDocument document;
    public PowerGridDocument doc() {return document;}

    @Click
    void btnOptions() {
        PowerGridOptionsActivity_.intent(this).start();
    }

    protected void resumeGame() {
        doc().resumeGame();
        PowerGridGameActivity_.intent(this).start();
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/powergrid/android/PowerGridMainActivity.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.androidimport

import com.zwstudio.logicpuzzlesandroid.R
import com.zwstudio.logicpuzzlesandroid.common.android.GameMainActivity
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.data.PowerGridDocument
import com.zwstudio.logicpuzzlesandroid.puzzles.powergrid.domain.PowerGridGame
import org.androidannotations.annotations.Bean
import org.androidannotations.annotations.EActivity

@EActivity(R.layout.activity_game_main)
class PowerGridMainActivity : GameMainActivity<PowerGridGame, PowerGridDocument, PowerGridGameMove, PowerGridGameState>() {
    @Bean
    protected lateinit var document: PowerGridDocument
    override fun doc() = document

    @Click
    fun btnOptions() {
        PowerGridOptionsActivity_.intent(this).start()
    }

    protected override fun resumeGame() {
        doc().resumeGame()
        PowerGridGameActivity_.intent(this).start()
    }
}

@EActivity(R.layout.activity_game_options)
class PowerGridOptionsActivity : GameOptionsActivity<PowerGridGame, PowerGridDocument, PowerGridGameMove, PowerGridGameState>() {
    @Bean
    protected lateinit var document: PowerGridDocument
    override fun doc() = document
}

@EActivity(R.layout.activity_game_help)
class PowerGridHelpActivity : GameHelpActivity<PowerGridGame, PowerGridDocument, PowerGridGameMove, PowerGridGameState>() {
    @Bean
    protected lateinit var document: PowerGridDocument
    override fun doc() = document
}